ibm/resource_ibm_tg_gateway.go

```diff
@@ -0,0 +1,417 @@
+			},
+
+			ResourceName: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The name of the resource",
+			},
+
+			ResourceCRN: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The crn of the resource",
+			},
+
+			ResourceStatus: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The status of the resource",
+			},
+
+			ResourceGroupName: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The resource group name in which resource is provisioned",
+			},
+		},
+	}
+}
+
+func resourceIBMTGValidator() *ResourceValidator {
+
+	validateSchema := make([]ValidateSchema, 1)
+
+	validateSchema = append(validateSchema,
+		ValidateSchema{
+			Identifier:                 tgName,
+			ValidateFunctionIdentifier: ValidateRegexpLen,
+			Type:                       TypeString,
+			Required:                   true,
+			AllowedValues:              `^([a-zA-Z]|[a-zA-Z][-_a-zA-Z0-9]*[a-zA-Z0-9])$}`,
+			MinValueLength:             1,
+			MaxValueLength:             63})
+
+	ibmTGResourceValidator := ResourceValidator{ResourceName: "ibm_tg_gateway", Schema: validateSchema}
+	return &ibmTGResourceValidator
+}
+
+func transitgatewayClient(meta interface{}) (*transitgatewayapisv1.TransitGatewayApIsV1, error) {
+	sess, err := meta.(ClientSession).TransitGatewayV1API()
+	return sess, err
+}
+
+func resourceIBMTransitGatewayCreate(d *schema.ResourceData, meta interface{}) error {
+	client, err := transitgatewayClient(meta)
+	if err != nil {
+		return err
+	}
+
+	location := d.Get(tgLocation).(string)
+	name := d.Get(tgName).(string)
+	global := d.Get(tgGlobal).(bool)
+
+	createTransitGatewayOptions := &transitgatewayapisv1.CreateTransitGatewayOptions{}
+
+	createTransitGatewayOptions.Name = &name
+	createTransitGatewayOptions.Location = &location
+	createTransitGatewayOptions.Global = &global
+
+	if _, ok := d.GetOk(tgResourceGroup); ok {
+		resourceGroup := d.Get(tgResourceGroup).(string)
+		createTransitGatewayOptions.ResourceGroup = &transitgatewayapisv1.ResourceGroupIdentity{ID: &resourceGroup}
+	}
+
+	//log.Println("going to create tgw now with options", *createTransitGatewayOptions.ResourceGroup)
+	tgw, response, err := client.CreateTransitGateway(createTransitGatewayOptions)
+	log.Println("Creation Success")
+
+	if err != nil {
+		log.Printf("[DEBUG] Create Transit Gateway err %s\n%s", err, response)
+		return err
+	}
+	d.SetId(*tgw.ID)
+
+	_, err = isWaitForTransitGatewayAvailable(client, d.Id(), d.Timeout(schema.TimeoutCreate))
+
+	v := os.Getenv("IC_ENV_TAGS")
+	if _, ok := d.GetOk(tgGatewayTags); ok || v != "" {
+		oldList, newList := d.GetChange(tgGatewayTags)
+		err = UpdateTagsUsingCRN(oldList, newList, meta, *tgw.Crn)
+		if err != nil {
+			log.Printf(
+				"Error on create of transit gateway (%s) tags: %s", d.Id(), err)
+		}
+	}
+	return resourceIBMTransitGatewayUpdate(d, meta)
```

Comment:
change return to resourceIBMTransitGatewayRead(d, meta) (We are not updating anything here after creation)

Reply:
ok.. changed.